src/ParseServer.js

```diff
@@ -168,7 +172,7 @@ class ParseServer {
 
     api.use('/health', function (req, res) {
       res.json({
-        status: 'ok',
+        status: options.state,
```

Comment:
This should be well documented; which states are possible and what does each state mean?

Reply:
Currently:

`ok`: server has started and is running
`error`: an error happened on startup (developer will need to check logs for details)
`initialized`: server has been created, `start` has not been called yet
`starting`: server is starting

Reply:
Nice! Could you add this to the Parse Server options definitions to document it?

Reply:
Can do!

Reply:
Is this the first time we're introducing more status codes, other than "ok"?

Reply:
Yep, currently it returns `ok` as soon as `new ParseServer` is called, even if there is an error in init, for example

Reply:
Let's think about how we want to use `status` in the future. I see 2 concepts:
a) status is a final state *only* related to the server start
b) status is a dynamic state that can change at anytime during runtime

I think (a) will make the status more versatile for future server health features. For example, if the server starts to encounter intermittent connection issues with the DB, it could change the status from `ok` to `warning`. Maybe in the future it could change to this during runtime:

```
{
  status: "warning",
  code: 1,
  warnings: [
    {
      code: 123,
      message: "Intermittent database connection errors"
    }
  ]
}
```

Reply:
Since Parse Server can go from "ok" to other status like "info" and still be up and healthy, this would be a perfect change for a major release. We could document all the other states already now to reserve the concept and require developers to adapt their existing health checks.

Reply:
That sounds like a good improvement to me. I'm pretty sure the code changes here will allow for the `health` to be updated by setting `config.state` (perhaps `config.health` is a more intuitive option). So, anywhere throughout Parse Server, health can easily be updated by updating `config`.

Reply:
We'll just add 4 unused `status` states to an enum, everything else is already in this PR it seems?

Reply:
I think there is a conceptual flaw in https://github.com/parse-community/parse-server/pull/8232#discussion_r996530923. If the status is `info`, the server would still be healthy, so suddenly one has to track multiple codes and interpret the health status from that. This needs some more thought...

Reply:
Not really sure how the `severe` endpoint would work too, if the express app/process crashes, the `/health` request would just return `404`